cli/class-wp-cli-simplechart.php

```diff
@@ -0,0 +1,211 @@
+	 *
+	 * ## OPTIONS
+	 * --ids=<ids>
+	 * : Comma-separated list of post IDs to migrate, or "all"
+	 *
+	 * [--dry-run]
+	 * : Simulate migration without updating database
+	 *
+	 * [--revert]
+	 * : Revert migration from Simplechart back to Chartbuilder
+	 *
+	 * @synopsis --ids=<ids> [--dry-run] [--revert]
+	 *
+	 * @subcommand migrate
+	 */
+	public function migrate_chartbuilder( $args, $assoc_args ) {
+		$this->setup_migration_opts( $assoc_args );
+
+		// Migrate all charts
+		if ( 'all' === $assoc_args['ids'] ) {
+			$this->loop_batch( array(
+				'post_status' => 'any',
+				'ignore_sticky_posts' => true,
+				'posts_per_page' => $this->posts_per_batch,
+				'post_type' => $this->post_type_from,
+				'orderby' => 'ID',
+			), 'migrate_single_post' );
+		}
+		// Migrate from specific list of IDs
+		else {
+			$ids = explode( ',', $assoc_args['ids'] );
+			if ( count( $ids ) > $this->posts_per_batch ) {
+				WP_CLI::error( sprintf( __( 'Max of %s ids can be specified, or use --ids=all', 'simplechart' ), $this->posts_per_batch ) );
+			}
+			foreach( $ids as $id ) {
+				$this->migrate_single_post( absint( $id ) );
+			}
+		}
+		WP_CLI::line( __( 'Migration complete', 'simplechart' ) );
+	}
+
+	/**
+	 * Migrates a single chart to Simplechart from the legacy Chartbuilder plugin
+	 *
+	 * ## OPTIONS
+	 * <id>
+	 * : Single post ID to migrate
+	 *
+	 * [--dry-run]
+	 * : Simulate migration without updating database
+	 *
+	 * [--revert]
+	 * : Revert migration from Simplechart back to Chartbuilder
+	 *
+	 * @synopsis <id> [--dry-run] [--revert]
+	 *
+	 * @subcommand migrate-single
+	 */
+	public function migrate_single_post( $args, $assoc_args = false ) {
+		// will be skipped if calling from $this->loop_batch
+		if ( $assoc_args ) {
+			$this->setup_migration_opts( $assoc_args );
+		}
+
+		if ( 'WP_Post' === get_class( $args ) ) {
+			$post = $args;
+		} else if ( ! empty( $args[0] ) && is_numeric( $args[0] ) ) {
+			$post = get_post( $args[0] );
+		} else {
+			WP_CLI::error( __( '`migrate_single_post()` requires a numeric ID or a WP_Post object', 'simplechart' ) );
+		}
+
+		if ( empty( $post ) ) {
+			WP_CLI::warning( sprintf( __( 'Could not locate post %s', 'simplechart' ), strval( $post ) ) );
+			return;
+		}
+
+		// confirm post type
+		if ( $this->post_type_from !== $post->post_type ) {
+			WP_CLI::warning( sprintf( __( 'Attempted to migrate post %s from invalid post_type "%s"', 'simplechart' ), $post->ID, $post->post_type ) );
+			return;
+		}
+
+		// confirm status
+		if ( wp_is_post_revision( $post->ID ) || 'auto-draft' === $post->post_status || 'inherit' === $post->post_status ) {
+			WP_CLI::warning( sprintf( __( 'Attempted to migrate post %s with invalid post_status "%s"', 'simplechart' ), $post->ID, $post->post_status ) );
+			return;
+		}
+
+		WP_CLI::line( sprintf( __( 'Updating post %s', 'simplechart' ), $post->ID ) );
+
+		if ( $this->dry_run ) {
```

Comment:
I feel like dry run should give some information about the dry run. What about putting checks for this around the update post meta calls and outputting a line message containing the posts that would be affected and/or the direction (to and from which post types). Giving the user some feedback on a dry run would inform any potential usage mistakes whereas leaving out any important feedback regarding the parameters of the command they issued could lead to a dangerous situation.